### PR TITLE
Release 4.0.0 PR

### PR DESCRIPTION
**Pull request by katmsft**

### CODE REVIEW

File: Changelog.txt [hunk 1]
```diff
@@ -1,6 +1,14 @@
 Azure Storage Client Library for C++
 History of Changes
 
+Changes in v4.0.0:
+- Fixed an issue where blob names that only contains space cannot be listed properly.
+- Added more compiler setting to unblock ApiScanning.
+- Added a new `nuspec` set that will release `Microsoft.Azure.Storage.CPP` and modified the `wastorage`'s nuspec to depend on Microsoft.Azure.Storage.CPP. Note that 3.3.0 will be the last version of `wastorage`. We strongly advice our user to use Nuget package `Microsoft.Azure.Storage.CPP` start from 3.3.0
```
Comment (JasonYang-MSFT) [inline Changelog.txt]: 4.0.0?

File: Microsoft.WindowsAzure.Storage/CMakeLists.txt [hunk 2]
```diff
@@ -107,16 +104,16 @@ set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/Binaries)
 set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/Binaries)
 
 set(AZURESTORAGE_INCLUDE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/includes)
-set(AZURESTORAGE_INCLUDE_DIRS ${CMAKE_CURRENT_SOURCE_DIR}/includes ${CASABLANCA_INCLUDE_DIRS} ${Boost_INCLUDE_DIRS} ${OPENSSL_INCLUDE_DIRS} ${LibXML++_INCLUDE_DIRS} ${UUID_INCLUDE_DIRS} ${Glibmm_INCLUDE_DIRS})
+set(AZURESTORAGE_INCLUDE_DIRS ${CMAKE_CURRENT_SOURCE_DIR}/includes ${CASABLANCA_INCLUDE_DIR} ${Boost_INCLUDE_DIRS} ${OPENSSL_INCLUDE_DIRS} ${UUID_INCLUDE_DIRS} ${LibXML2_INCLUDE_DIR})
```
Comment (JasonYang-MSFT) [inline Microsoft.WindowsAzure.Storage/CMakeLists.txt]: CASABLANCA_INCLUDE_DIR? Any reason for this change?

Comment (katmsft) [reply Microsoft.WindowsAzure.Storage/CMakeLists.txt]: Yes, when compile on linux I found that using DIRS cannot locate Casablanca in some cases. Hence the change.

File: Changelog.txt (hunk 1 above)
Comment (katmsft) [reply Changelog.txt]: Nice catch, will modify it.

File: Microsoft.WindowsAzure.Storage/CMakeLists.txt (hunk 2 above)
Comment (JasonYang-MSFT) [reply Microsoft.WindowsAzure.Storage/CMakeLists.txt]: Have you test is change on vcpkg?

Comment (katmsft) [reply Microsoft.WindowsAzure.Storage/CMakeLists.txt]: Nope, I'll try to test it out first.

**Pull request closed without merging**